aarch64_qemuvirt/src/main.rs

```diff
@@ -21,14 +17,10 @@ const DTB_ADDR: usize = 0x4000_0000;
 extern "C" fn k_main(_device_tree_ptr: usize) -> ! {
     kernel::arch::aarch64::Aarch64::disable_fp_trap();
 
-    kernel::kernel_console::init(Pl011::new(0x0900_0000));
-    kernel::kprintln!("hello!");
+    static PL011: Pl011 = Pl011::new(0x0900_0000);
+    kernel::globals::set_console(&PL011);
```

Comment:
Couldn't this be moved to the global instead of declaring a static here?

Reply:
What do you mean ? Moving the PL011 variable to global scope ? Or directly creating in the global ?
The latter wouldn't be possible because at compile time, we can't always know the base address of the console, therefore it has to be created at runtime)

Reply:
I was talking about something like that:
```suggestion
    let pl011 = Pl011::new(0x0900_0000);
    kernel::globals::set_console(pl011);
```

Reply:
A local then.
I used a static so that it wouldn't take up storage space on the stack. Our kernel's stack is a scarce resource.

Reply:
Ok I can see that